docs/developer-guide.md

```diff
@@ -43,6 +43,20 @@ You can undeploy Katib v1beta1 manifests from a Kubernetes cluster as follows:
 make undeploy
 ```
 
+## Technical and style guide
+
+The following guidelines apply primarily to Katib, 
+but other projects like training-operator might also adhere to them. 
+
```

Comment:
Should we make a section about Golang here since Katib has various components written in Go, Python, Typescript?


```suggestion

## Golang Development

```

Reply:
SGTM

Reply:
Done.